source/Camera/InfoUtilOther/ZividBenchmark/ZividBenchmark.cpp

```diff
@@ -0,0 +1,453 @@
+#include <Zivid/Zivid.h>
+#include <algorithm>
+#include <iostream>
+
+namespace
+{
+    const int PRINT_WIDTH = 56;
+
+    using HighResClock = std::chrono::high_resolution_clock;
+    using Duration = std::chrono::nanoseconds;
+
+    Duration computeAverageDuration(const std::vector<Duration> &durations)
+    {
+        return std::accumulate(durations.begin(), durations.end(), Duration{ 0 }) / durations.size();
+    }
+
+    Duration computeMedianDuration(std::vector<Duration> durations)
+    {
+        std::sort(durations.begin(), durations.end());
+        if(durations.size() % 2 == 0)
+        {
+            return (durations.at(durations.size() / 2 - 1) + durations.at(durations.size() / 2)) / 2;
+        }
+        else
+        {
+            return durations.at(durations.size() / 2);
+        }
+    }
+
+    template<typename T>
+    std::string valueToStringWithPrecision(const T &value, const size_t precision)
+    {
+        std::ostringstream ss;
+        ss << std::setprecision(precision) << std::fixed << value;
+        return ss.str();
+    }
+
+    std::string formatDuration(const Duration &duration)
+    {
+        return valueToStringWithPrecision(
+                   std::chrono::duration_cast<std::chrono::duration<double, std::milli>>(duration).count(), 3)
+               + " ms";
+    }
+
+    template<typename Target>
+    std::string makeSettingList(const std::vector<Zivid::Settings> &settingsVector)
+    {
+        std::string settingList = "{ ";
+        for(size_t i = 0; i < settingsVector.size(); i++)
+        {
+            settingList += settingsVector.at(i).get<Target>().toString();
+            if(i + 1 != settingsVector.size()) settingList += ", ";
+        }
+        settingList += " }";
+        return settingList;
+    }
+
+    std::string makefilterList(const std::vector<Zivid::Settings> &settingsVector)
+    {
+        if(settingsVector.at(0).filters().gaussian().isEnabled().value())
+        {
+            if(settingsVector.at(0).filters().reflection().isEnabled().value())
+            {
+                return "{ gaussian, reflection }";
+            }
+            else
```

Comment:
Else after return is redundant, I think I proposed this before. You think this is better?

Reply:
Maybe I missed it, or didn't understand what you meant. I got rid of the elses I understand are redundant now.